x509/src/pkix_extensions.rs

```diff
@@ -0,0 +1,1041 @@
+//! Extensions [`Extensions`] as defined in RFC 5280
+
+use crate::certificate::default_false;
+use crate::general_name::GeneralName;
+use crate::general_name::GeneralNames;
+use crate::RelativeDistinguishedName;
+use crate::{default_zero, AttributeTypeAndValue};
+use der::asn1::{
+    Any, BitString, ContextSpecific, GeneralizedTime, Ia5String, Null, ObjectIdentifier,
+    OctetString, UIntBytes, Utf8String,
+};
+use der::{
+    Choice, Decodable, DecodeValue, Decoder, Encodable, EncodeValue, Enumerated, ErrorKind,
+    FixedTag, Length, Sequence, Tag, TagMode, TagNumber,
+};
+
+/// DisplayText as defined in [RFC 5280 Section 4.2.1.4] in support of the Certificate Policies extension.
+///
+/// ASN.1 structure for DisplayText is below. At present, only the ia5String and utf8String options are supported.
+///
+/// ```text
+///    DisplayText ::= CHOICE {
+///         ia5String        IA5String      (SIZE (1..200)),
+///         visibleString    VisibleString  (SIZE (1..200)),
+///         bmpString        BMPString      (SIZE (1..200)),
+///         utf8String       UTF8String     (SIZE (1..200)) }
+/// ```
+///
+/// [RFC 5280 Section 4.2.1.4]: https://datatracker.ietf.org/doc/html/rfc5280#section-4.2.1.4
+#[derive(Choice, Clone, Debug, Eq, PartialEq)]
+pub enum DisplayText<'a> {
+    /// ia5String        IA5String      (SIZE (1..200))
+    #[asn1(type = "IA5String")]
+    Ia5String(Ia5String<'a>),
+
+    /// visibleString    VisibleString  (SIZE (1..200)),
+    // TODO: support VisibleString if desired
+
+    /// bmpString        BMPString      (SIZE (1..200)),
+    // TODO: support BMPString if desired
+
+    /// utf8String       UTF8String     (SIZE (1..200))
+    #[asn1(type = "UTF8String")]
+    Utf8String(Utf8String<'a>),
+}
+
+/// Extended key usage extension as defined in [RFC 5280 Section 4.2.1.12] and as identified by the [`PKIX_CE_EXTKEYUSAGE`](constant.PKIX_CE_EXTKEYUSAGE.html) OID.
+///
+/// Many extended key usage values include:
+/// - [`PKIX_CE_ANYEXTENDEDKEYUSAGE`](constant.PKIX_CE_ANYEXTENDEDKEYUSAGE.html),
+/// - [`PKIX_KP_SERVERAUTH`](constant.PKIX_KP_SERVERAUTH.html),
+/// - [`PKIX_KP_CLIENTAUTH`](constant.PKIX_KP_CLIENTAUTH.html),
+/// - [`PKIX_KP_CODESIGNING`](constant.PKIX_KP_CODESIGNING.html),
+/// - [`PKIX_KP_EMAILPROTECTION`](constant.PKIX_KP_EMAILPROTECTION.html),
+/// - [`PKIX_KP_TIMESTAMPING`](constant.PKIX_KP_TIMESTAMPING.html),
+///
+/// ```text
+/// ExtKeyUsageSyntax ::= SEQUENCE SIZE (1..MAX) OF KeyPurposeId
+/// ```
+///
+/// [RFC 5280 Section 4.2.1.12]: https://datatracker.ietf.org/doc/html/rfc5280#section-4.2.1.12
+pub type ExtendedKeyUsage<'a> = alloc::vec::Vec<ObjectIdentifier>;
```

Comment:
Would suggest a `use alloc::vec::Vec` import at the top of the module, especially since there are multiple `Vec`s declared (e.g. SubjectDirectoryAttributes as well)

Reply:
Done (sorry about this one, meant to make that mod after the same comment on general_name and lost that thread today).